blockchain/v2/scheduler.go

```diff
@@ -372,15 +365,9 @@ func (sc *scheduler) setStateAtHeight(height int64, state blockState) {
 	sc.blockStates[height] = state
 }
 
+// CONTRACT: peer exists and in Ready state.
```

Comment:
Should we really assume this? Concurrency is hard to reason about, isn't it optimistic to think we'll be able to enforce this?

Reply:
It isn't about concurrency. If you look at the code, you'll see that `touchPeer` func is called before this func. `touchPeer` ensures that the peer exists.